lib/adf_diag.py

```diff
@@ -1080,3 +1108,227 @@ def derive_variables(self, vars_to_derive=None, ts_dir=None, overwrite=None):
+                            print(
+                                f"ERROR: Necessary 'time_period_freq' attribute missing from {adf_file}"
+                            )
+                        continue
+
+                    found_strings = [
+                        word for word in freq_string_options if word in dataset_freq
+                    ]
+                    if len(found_strings) == 1:
+                        if verbose > 2:
+                            print(
+                                f"Found dataset_freq {dataset_freq} matches {found_strings}"
+                            )
+                    elif len(found_strings) > 1:
+                        if verbose > 0:
+                            print(
+                                f"WARNING: Found dataset_freq {dataset_freq} matches multiple string possibilities:{', '.join(found_strings)}"
+                            )
+                    else:
+                        if verbose > 0:
+                            print(
+                                "ERROR: None of the frequency options {freq_string_options} are present in the time_period_freq attribute {time_perido_freq}"
+                            )
+                            print("Skipping {adf_file}")
+                            freq = "frequency_missing"
+                        continue
+                    freq = found_strings[0]
+
+                    #
+                    # Destination file is MDTF directory and name structure
+                    #
+                    mdtf_dir = os.sep.join([mdtf_model_data_root, case_name, freq])
+
+                    os.makedirs(mdtf_dir, exist_ok=True)
+                    mdtf_file = (
+                        mdtf_dir + os.sep + ".".join([case_name, var, freq, "nc"])
+                    )
+                    mdtf_file_list = glob.glob(
+                        mdtf_file
+                    )  # Check if files already exist in MDTF directory
+                    if (
+                        mdtf_file_list
+                    ):  # If files exist, then should check if over-writing is allowed:
+                        if verbose > 0:
+                            print(
+                                f"\t   INFO: not clobbering existing mdtf file {mdtf_file_list}"
+                            )
+                        # logic to allow override#  if not overwrite_mdtf[case_idx]:
+                        continue  # simply skip file copy for this variable:
+
+                    if verbose > 1:
+                        print(f"copying {adf_file} to {mdtf_file}")
+                    shutil.copyfile(adf_file, mdtf_file)
+                # end for hist_str
+            # end for var
+        # end for case
+
+    def setup_run_mdtf(self):
+        """
+        Create MDTF directory tree, generate input setttings jsonc file
+        Submit MDTF diagnostics.
+
+        """
+
+        # import needed standard modules:
+        import json
```

Comment:
I might move this to the top of `adf_diag.py` where the rest of the import statements are.